enhancements/profile-versioning.md

```diff
@@ -0,0 +1,413 @@
+more complex version strings that use pre-release or build metadata within the
+semantic version specification to denote changes to a profile between operator
+releases. If we do decide to support pre-release or build metadata as part of
+the version string, we should ensure version string evaluation considers
+`1.2.0` older than `1.2.0-1` or `1.2.0-2+e076b1d7ac`. This detail will be
+covered in a future enhancement if we decide to implement granular profile
+versions, as noted in the non-goals section above.
+
+Note that we can't enforce semantic versioning because the version is
+ultimately set by the authoring body of the standard.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There are two considerations for how versions interact with profiles.
+
+The first is if there is only one profile, and it has a single version. For
+example, supporting the CIS OpenShift 1.4.0 benchmark. The second is if there
+is a single profile, but multiple versions. For example, PCI-DSS version 3.2.1
+and version 4.0.0.
+
+#### Single Profile & Single Version
+
+This case can be implemented in the following steps:
+
+1. Associate a version to the profile in
+   [ComplianceAsCode/content](https://github.com/ComplianceAsCode/content)
+   using the `metadata.version` attribute of the profile
+2. Update the operator's profile parser to parse the version from the data
+   stream, and set the appropriate attribute on the `Profile` custom resource
+
+This will look similar to the following after the profiles are parsed by the
+operator on start-up:
+
+```console
+$ oc get profiles.compliance
+NAME       AGE     VERSION
+pci-dss    2d      3.2.1
+ocp4-cis   2d      1.4.0
+```
+
+Profiles in the
+[ComplianceAsCode/content](https://github.com/ComplianceAsCode/content) that do
+not have a `metadata.version`, will not have that attribute set in the
+`Profile` custom resource (using `ocp4-nerc-cip` purely as an example below):
+
+```console
+$ oc get profiles.compliance
+NAME           AGE   VERSION
+pci-dss        2d    3.2.1
+ocp4-cis       2d    1.4.0
+ocp4-nerc-cip  2d
+```
+
+#### Single Profile & Multiple Versions
+
+The operator must account for cases where a profile has multiple versions.
+
+Due to limitations of the data stream, each profile can only have one version.
+Additionally, there can only be one profile using a given name in the data
+stream.
+
+The operator can expose multiple versions for a single profile using the following approach:
+
+1. Version each profile to the initially supported version, keeping the profile name the same
+2. When a new version is released, implement it using the versionless profile named
+3. Implement a new profile for the older version under a new name
+
+For example, this would look like the following in practice:
+
+1. Version the current CIS OpenShift benchmark at version 1.4.0
+2. When version 1.5.0 is implemented, move version 1.4.0 to a new profile named `ocp4-cis-1.4.0`)
```

Comment:
This is a good point, although I am not sure how to test this point now.

Reply:
I added some details below on how we can test this in upgrade scenarios.